Provided a hack to match the video size to the poster image size better.

eklenske opened the pull request:
I am not sure about the origin of the problem, but when I use videos, they don't match the poster image perfectly in size, leading to flickering effects.

I changed the size of the video area to prevent this flickering. Now the videos are nicely aligned with the poster image.

Maybe someone figures out the origin of the problem (does poppler add a margin around link object areas?), but until then, this hack should work for most people.

AndreasBilke commented:
Could you provide some example-video where the flickering happens? I am a
litte careful with these "fixed-number-hacks" since they can be a bad idea
with different dpi settings of the monitor.

I just want to test your changes against different dpi settings.

eklenske commented:
Sure, here is one [example presentation with videos](https://owncloud.tuebingen.mpg.de/public.php?service=files&t=3db0107ab6adcec6b8af17e2f3888c19), where I experience a size difference between poster image and video.

I totally agree with you about being careful with fixed-number-hacks. That's why I made it clear that this is just a hack. If the problem could be fixed upstream, it would be much better, of course. But until then, this hack works for me. :)

AndreasBilke commented:
Could you also post the tex code?

For me, the videos are not played at all in pdfpc, would you mind if we use these for issue #90 as an example?

eklenske commented:
Sure, I changed the [example presentation](https://owncloud.tuebingen.mpg.de/public.php?service=files&t=3db0107ab6adcec6b8af17e2f3888c19) to an MWE and added the tex source as well. Flipping back and forth between the slides should show you the size difference between poster and video (if the video works at all, of course).

Of course you can also use it for other issues. (Disclaimer: I had issues with some of my videos not playing on newer Ubuntu versions, but everything works on 14.04LTS. I am not sure if this video is one with issues or not, maybe it just works.)

AndreasBilke commented:
Okay I see the problem. Does this happen to every of your videos? I also will try some.

eklenske commented:
Yes, this happens to all of my videos, no matter if they are created with Matlab or ffmpeg.

AndreasBilke commented:
I tried some other videos by myself. What I discovered is, that if you use hyperref with
```
\href{run:trailer.mkv}{\includegraphics[width=8cm, height=4.5cm]{trailer.png}}
```

stuff, I also get the size-problems. If I start pdfpc I get the right size if the image preview, but a moment later this preview shrinks.

But now to the important part: If I use the multimedia package of beamer and insert a video with
```
\movie{\includegraphics[width=8cm, height=4.5cm]{trailer.png}}{trailer.mkv}
```

everything is fine.

So your pull request might destroy embedded videos which comes with the multimedia package. I have no idea so far, why this happens :-)

AndreasBilke commented:
I looked into the pdf to find out what is different between both solutions. Since I have no clue about internals of pdf I can only show you what I saw:

I created two pdfs with no difference except the hyperref and the movie solutions. This is the part, where both solutions differ. It seems that this is the annotation area (the rect).

```
%% Original object ID: 13 0
11 0 obj
<<
  /A <<
    /F (trailer.mkv)
    /S /Launch
  >>
  /Border [
    0
    0
    0
  ]
  /C [
    0
    .7
    .7
  ]
  /H /N
  /Rect [
    68.039
    87.742
    294.796
    215.292
  ]
  /Subtype /Link
  /Type /Annot
>>
endobj
```

hyperref produces a bigger rect (for the same file) than movie. If I copy the rect size from movie to hyperref, everything is fine. No rezising.

I can only guess what happens. Maybe hyperref makes some margins around the preview image and pdfpc draws the video within the annotation area (which is bigger than the actual image).

AndreasBilke commented:
Okay now I am sure that hyperref makes some margins around the image. This behaviour is nice if you have links on text (to increase the clickable area) but for the movie case this is rather bad.

I also see no possibility to fix this in pdfpc. We use poppler to get the clickable area and I see no way in auto detecting hyperref links.

Maybe there is a way to tell hyperref directly to skip the extra margins in latex, but that is a hyperref thing and not pdfpc.

eklenske commented:
Thanks for figuring this out!
